src/attach-shadow.js

```diff
@@ -521,3 +551,75 @@ export function attachShadow(host, options) {
   }
   return new ShadyRoot(ShadyRootConstructionToken, host, options);
 }
+
+// Mitigate connect/disconnect spam by wrapping custom element classes.
+if (window['customElements']) {
+
+  // process connect/disconnect after roots have rendered to avoid
+  // issues with reaction stack.
+  let connectMap = new Map();
+  rootRendered = function() {
```

Comment:
Noting: This stuff would all go away with the "append/remove" synchronously (then distribute & squelch later)" gambit.